### PR TITLE
Fixes and improvements on menus

### PR DESCRIPTION
**Pull request by m4dz**
This PR mainly fix two things:
- the drawer than was below footer and breadcrumb in mobile
- icons that wasn't properly loaded

It also brings some improvement with libs and code structure to make them testable (incoming in future PR)

### CODE REVIEW

File: src/components/Drawer.svelte [hunk 1]
```diff
@@ -22,19 +22,52 @@
...
+       * We manage the [aria-hidden] attribute manually, as it serves for CSS
+       * transitions, and needs to be wrapped in next frames ticks to ensure
+       * smooth movements.
+       */
+      this.refs.wrapper.setAttribute('aria-hidden', !this.get('visible'))
+
+      /**
+       * Animation engine, based on CSS transitions
+       *
+       * This is how it works :
```
Comment (CPatchane) [inline src/components/Drawer.svelte]: 👍

File: src/styles/bar.css [hunk 2]
```diff
@@ -63,8 +70,21 @@
   text-indent: -9999em;
```
Comment (CPatchane) [inline src/styles/bar.css]: Ooh, why so much?

File: src/styles/bar.css [hunk 3]
```diff
@@ -63,8 +70,21 @@
...
-  [role=banner] .coz-bar-burger {
+@media (min-width: 48.0625em) {
+  [role=banner] .coz-bar-burger,
+  [role=banner] .coz-drawer-wrapper {
+    /* Fix unwanted `!important` in
+     * https://github.com/cozy/cozy-ui/blob/v3/stylus/ui-base/boxes.styl#L9
+     *
+     * FIXME: remove it when hide() method is removed from cozy-ui
+     */
+    display: none !important;
```
Comment (CPatchane) [inline src/styles/bar.css]: It's now merged in cozy-ui, with the next release I think that you can update this part.

File: src/styles/drawer.css [hunk 4]
```diff
@@ -4,16 +4,24 @@
...
   width: 100vw;
   height: 100vh;
+  /* Prepare for transitions */
+  display: none;
+  /* Fix unwanted `!important` in
+   * https://github.com/cozy/cozy-ui/blob/v3/stylus/ui-base/boxes.styl#L9
+   *
+   * FIXME: remove it when hide() method is removed from cozy-ui
+   */
+  visibility: visible;
```
Comment (CPatchane) [inline src/styles/drawer.css]: Again here (cozy-ui release)

File: src/styles/drawer.css [hunk 5]
```diff
@@ -4,16 +4,24 @@
...
-[role=banner] .coz-drawer-wrapper[aria-hidden=true] {
-  display: none;
+[role=banner] .coz-drawer-wrapper[aria-hidden=false],
+[role=banner] .coz-drawer-wrapper.swipe-active {
+  /* Fix unwanted `!important` in
+   * https://github.com/cozy/cozy-ui/blob/v3/stylus/ui-base/boxes.styl#L9
+   *
+   * FIXME: remove it when hide() method is removed from cozy-ui
+   */
+  display: block !important;
```
Comment (CPatchane) [inline src/styles/drawer.css]: Again here (cozy-ui release)

File: src/styles/bar.css (hunk 2 above)
Comment (m4dz) [inline src/styles/bar.css]: Only to  push the text content outside of the viewport and keep the place for the icon. It was the best practice for a long time, but maybe @GoOz knows a better way to do this?

**Comment by m4dz** [comment]
> include the next cozy-ui release changes 

Yup :)

File: src/components/Bar.svelte [hunk 6]
```diff
@@ -138,18 +134,61 @@
     return clone
   }
 
+  /**
+   * Helper function to update apps in CONFIG tree
+   * @param  {Object}           config the JSON CONFIG tree source
+   * @return {Promise(boolean)} a valve that allow to trigger update or not
```
Comment (lespacedunmatin) [inline src/components/Bar.svelte]: s/valve/value je pense.

Comment (m4dz) [inline src/components/Bar.svelte]: Nope, it returns a `valve` in functionnal term: to boolean that allow to block or pass a process later :smiley: (see https://github.com/cozy/cozy-bar/pull/28/files/6ff3b3fa99fe41f76933bf0f945073caac793954#diff-c8a245b0e389919c587f61c8d66df421R230 for some kind of use)

File: src/styles/bar.css (hunk 2 above)
Comment (GoOz) [inline src/styles/bar.css]: I wonder if it's not a good case for using `$hidden` or `.coz-hidden` 
https://github.com/cozy/cozy-ui/blob/v3/stylus/ui-base/mixins.styl#L25

File: src/styles/bar.css (hunk 3 above)
Comment (GoOz) [inline src/styles/bar.css]: Incoming… like in a few minutes.

File: src/index.js [hunk 7]
```diff
@@ -32,25 +32,55 @@ const injectDOM = function CozyBarInjectDOM (data) {
...
     data
   })
 }
 
 const bindEvents = function CozyBarBindEvents () {
+  const body = document.body
+  const root = document.querySelector('[role=banner]')
+  const aside = document.querySelector('.coz-drawer-wrapper aside')
+
+  /** Fire a `clickOutside` event when clcking everywhere in the viewport */
```
Comment (lespacedunmatin) [inline src/index.js]: s/clcking everywhere/clicking anywhere/.

File: src/styles/bar.css (hunk 2 above)
Comment (m4dz) [inline src/styles/bar.css]: Technically, I agree. But I probably won't rely on `.coz-hidden` as it assumes the app uses cozy-ui, which we can't expect for future contributions. The Bar must stay standalone. Whatever, I can duplicate the mixin here :+1:

Comment (GoOz) [inline src/styles/bar.css]: Oh yeah, you're right. Indeed, just duplicate it then 😄

**Comment by enguerran** [comment]
*note: I reference https://github.com/cozy/cozy-files-v3/pull/182 in this PR to have its status displayed in the `cozy-files-v3` status as it depends on it.*

**Pull request merged**